src/decorators/types.ts

```diff
@@ -32,7 +32,7 @@ export interface ValidateOptions {
   validate?: boolean | ValidatorOptions;
 }
 export interface ComplexityOptions {
-  complexity?: Complexity;
+  complexity?: ((arg: ComplexityEstimatorArgs) => number) | number;
```

Comment:
There is a `ComplexityEstimator` type that we should use - `complexity` function can return `void` to fallback to default complexity